### PR TITLE
Dockerfile: pass BUILDTAGS to `go build`

### PR DESCRIPTION
**Pull request by kaovilai**

### CODE REVIEW

**Review by crazy-max — APPROVED** [review]

**Comment by thaJeztah** [comment]
Looks like there's an issue; perhaps quoting is incorrect? (haven't checked);

```
#23 0.636 malformed import path "-trimpath": leading dash
#23 0.636 malformed import path "-ldflags": leading dash
#23 0.636 malformed import path "-X github.com/distribution/distribution/v3/version.Version=2.7.0-2007-ge1a30f7d -X github.com/distribution/distribution/v3/version.Revision=e1a30f7d2c609a9ad2f73b3566dab69cf5c89298 -X github.com/distribution/distribution/v3/version.Package=github.com/distribution/distribution/v3 -s -w": leading dash
#23 0.636 malformed import path "-o": leading dash
```

File: Dockerfile
Comment (crazy-max) [inline Dockerfile:30]:
```suggestion
      set -x ; xx-go build -tags "${BUILDTAGS}" -trimpath -ldflags "$(cat /tmp/.ldflags) ${LDFLAGS}" -o /usr/bin/registry ./cmd/registry \
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
Per @thaJeztah comment, missing quotes.

File: Dockerfile
Comment (milosgajdos) [reply Dockerfile:30]: Commit suggestions screw with signatures 😄

Comment (crazy-max) [reply Dockerfile:30]: @milosgajdos Indeed thanks for the reminder :sweat_smile:

Comment (kaovilai) [reply Dockerfile:30]: noted

**Comment by kaovilai** [comment]
squahsing into 1 commit just in case

**Comment by kaovilai** [comment]
blocked by https://github.com/distribution/distribution/pull/3702. So cherry pick into there maybe?

**Comment by milosgajdos** [comment]
There is a whole chain of PRs 😬 I'd love someone with proper knowledge of GCS to review #3702 but I dont think we have anyone in @distribution/distribution-maintainers whos got that knowledge 🤔

**Comment by kaovilai** [comment]
From digging.. it was after https://github.com/distribution/distribution/commit/87f93ede9e5de6e86a2d2aefb2d5d8fe7ce98274#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L25 that tags were lost

**Comment by flavianmissi** [comment]
FTR there's a new PR up to fix the GCS driver (the previous one was closed): https://github.com/distribution/distribution/pull/3929

**Comment by kaovilai** [comment]
blocked by https://github.com/distribution/distribution/pull/3929

**Comment by milosgajdos** [comment]
There is also https://github.com/distribution/distribution/pull/3926

**Comment by kaovilai** [comment]
> There is a whole chain of PRs 😬 I'd love someone with proper knowledge of GCS to review https://github.com/distribution/distribution/pull/3702 but I dont think we have anyone in @distribution/distribution-maintainers whos got that knowledge 🤔

What's the plan with v3 if there is no GCS maintainer? add new maintainer? kill off GCS driver?

**Comment by DavidSpek** [comment]
@kaovilai Could you rebase this PR since https://github.com/distribution/distribution/pull/3929 has now been merged.

**Comment by milosgajdos** [comment]
Closing, addressed in https://github.com/distribution/distribution/pull/3950

**Pull request closed without merging**
